app/src/lib/stores/app-store.ts

```diff
@@ -1474,6 +1495,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
     repository: Repository,
     previouslySelectedRepository: Repository | CloningRepository | null
   ): Promise<Repository | null> {
+    this.refreshBranchProtectionState(repository)
+
+    // This is now purely for metrics collection for `commitsToRepositoryWithBranchProtections`
+    // Understanding how many users actually contribute to repos with branch protections gives us
+    // insight into who our users are and what kinds of work they do
+    this.updateBranchProtectionsFromAPI(repository)
```

Comment:
Now that we're only using the `repos/${owner}/${name}/branches?protected=true` endpoint to gather metrics about if a user is committing to a repository that has protected branches, we can afford to update the data less frequently (individual rule changes don't matter to us, only if the repo has a non-zero number of protected branches). So for now I'm opting to simply update it on repo change.

Reply:
💯!